spec/support/reporters/nyc-test-coverage-reporter.mjs

```diff
@@ -0,0 +1,32 @@
+// This file runs in node.js.
+
+import fsPromises from 'node:fs/promises';
+import {URL} from 'node:url';
+import path from 'node:path';
+
+export default class NycTestCoverageReporter {
+  // The following method runs after every Jasmine describe block.
+  suiteDone({properties}) {
+    this.coverage = properties?.__coverage__ || this.coverage;
+  }
+
+  jasmineDone() {
+    const rootDirname = path.dirname(
+      new URL(import.meta.resolve('#root/package.json')).pathname
+    );
+    const reportPath = path.join(rootDirname, 'coverage', 'javascript');
+    const nycOutputPath = path.join(rootDirname, '.nyc_output');
+    const assetsPath = path.join(nycOutputPath, 'assets');
+    const testCoveragePath = path.join(nycOutputPath, 'out.json');
+    return Promise.all([
+      /* Remove the previous report and the nyc-instrumented assets
+       * as they are not needed after the run has completed,
+       * and if not removed, they would accumulate and potentially
+       * confuse the developer.
+       */
+      fsPromises.rm(reportPath, {recursive: true, force: true}),
+      fsPromises.rm(assetsPath, {recursive: true, force: true}),
```

Comment:
After looking at this further we should be able to use the `delete` option in nyc to remove the output directory - 

> The --delete option will remove the existing output directory before instrumenting.

https://github.com/istanbuljs/nyc/blob/main/docs/instrument.md#pre-instrumented-source

For example, we would update this line - https://github.com/alphagov/frontend/pull/4142/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R12

From: `"nyc:instrument": "yarn run nyc instrument public/assets/frontend .nyc_output/assets"`

To: "nyc:instrument": "yarn run nyc instrument public/assets/frontend .nyc_output/assets --delete"

Reply:
A good idea, let me test it.

Reply:
Interestingly this options is not documented when one runs `yarn run nyc instrument --help` but it seems to work.

Reply:
Done